src/api/repos.rs

```diff
@@ -0,0 +1,173 @@
+                    "ref": reference.full_ref_url(),
+                    "sha": sha.into(),
+                })),
+            )
+            .await
+    }
+
+    /// Creates a new file in the repository.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// use octocrab::models::AuthorUser;
+    ///
+    /// // Commit to add "crabs/ferris.txt"
+    /// octocrab::instance()
+    ///     .repos("owner", "repo")
+    ///     .create_file(
+    ///         "crabs/ferris.txt",
+    ///         "Created ferris.txt",
+    ///         "Thought there’d never be a Rust Rap?\n"
+    ///     )
+    ///     .branch("master")
+    ///     .commiter(AuthorUser {
+    ///         name: "Octocat".to_string(),
+    ///         email: "octocat@github.com".to_string(),
+    ///     })
+    ///     .author(AuthorUser {
+    ///         name: "Ferris".to_string(),
+    ///         email: "ferris@rust-lang.org".to_string(),
+    ///     })
+    ///     .send()
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn create_file(
+        &self,
+        path: impl Into<String>,
+        message: impl Into<String>,
+        content: impl AsRef<[u8]>,
+    ) -> UpdateFileBuilder<'_, '_> {
+        UpdateFileBuilder::new(
+            self,
+            path.into(),
+            message.into(),
+            base64::encode(content),
+            None,
+        )
+    }
+
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// # let blob_sha = "";
+    /// use octocrab::models::AuthorUser;
+    ///
+    /// // Given the file blob for "crabs/ferris.txt", commit to update the file.
+    /// octocrab::instance()
+    ///     .repos("owner", "repo")
+    ///     .update_file(
+    ///         "crabs/ferris.txt",
+    ///         "Updated ferris.txt",
+    ///         "But me and Ferris Crab: best friends to the end.\n",
+    ///         blob_sha
+    ///     )
+    ///     .branch("master")
+    ///     .commiter(AuthorUser {
+    ///         name: "Octocat".to_string(),
+    ///         email: "octocat@github.com".to_string(),
+    ///     })
+    ///     .author(AuthorUser {
+    ///         name: "Ferris".to_string(),
+    ///         email: "ferris@rust-lang.org".to_string(),
+    ///     })
+    ///     .send()
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn update_file(
+        &self,
+        path: impl Into<String>,
+        message: impl Into<String>,
+        content: impl AsRef<[u8]>,
+        sha: impl Into<String>,
+    ) -> UpdateFileBuilder<'_, '_> {
+        UpdateFileBuilder::new(
+            self,
+            path.into(),
+            message.into(),
+            base64::encode(content),
+            Some(sha.into()),
+        )
+    }
+
+    /// Creates a new `PullRequestHandler` for the repository,
+    /// that allows you to access GitHub's pull request API.
+    pub fn pulls(&self) -> super::pulls::PullRequestHandler {
```

Comment:
Is there a reason to add this?

Reply:
Ease of use? I found it useful when writing my example of creating a file then opening a PR to the repository, so I added it, but I can remove it.

Reply:
For now I would prefer that, just to try to limit the API, and keep it similar to octokit. Once API is a lot more fleshed out, used, and stable, we can add more convenience methods.